plugin.py

```diff
@@ -35,10 +37,10 @@ def name(cls) -> str:
         return cls.__name__
 
     @classmethod
-    def configuration(cls) -> Tuple[sublime.Settings, str]:
-        settings, filename = super().configuration()
-        settings.set("command", cls.get_command())
-        return settings, filename
+    def get_settings(cls) -> sublime.Settings:
+        return sublime.load_settings(
+            "LSP-{}.sublime-settings".format(cls.name())
+        )
```

Comment:
I don't know...

I'd instead remove `get_settings`, `get_command`, `on_pre_start` and `get_x_command()` and have the platform-specific parameters controlled through variables defined in `additional_variables()` (`binary_path` and `binary_arguments` or something like that).

Reply:
but the command is completely different on Linux/macOS compared to Windows... not sure how you would unify that?

command on unix:

mono --assembly-loader=strict --config etc/config OmniSharp.exe --languageserver

command on Windows:

OmniSharp.exe --languageserver

Reply:
One variable for `$server_binary` and one for `$server_arguments`?

Reply:
ah, no, the arguments need to be in a list...

Reply:
But then wouldn't it work to do it like now but decide and return correct command from the `configuration()` implementation already?

Reply:
Possibly, but on_pre_start is exactly the callback we cooked up to set up the command at the last minute?

Reply:
If the `configuration()` implementation is not the most appropriate place to return the configuration then I'm not sure what is. :)

The `on_pre_start` is for cases when we have to decide based on view/window or other context that it provides but in this case there is no need for that.

Reply:
> If the configuration() implementation is not the most appropriate place to return the configuration then I'm not sure what is. :)

That's only for returning the settings object so LSP can add a settings change listener to it.